src/main/java/org/opentripplanner/updater/alert/TransitAlertProvider.java

```diff
@@ -2,6 +2,13 @@
 
 import org.opentripplanner.routing.services.TransitAlertService;
 
+/**
+ * Interface for things that maintain their own individual index associating TransitAlerts with the
+ * transit entities they affect. In practice, these are always realtime updaters handling GTFS-RT
+ * Alerts or Siri SX messages. This interface appears to exist only to allow merging multiple such
+ * services together, which appears to be a workaround for not maintaining snapshots of a single
+ * instance-wide index.
+ */
 public interface TransitAlertProvider {
```

Comment:
This interface is a workaround for the (to me, surprising) fact that each realtime update handler is maintaining its own independent mapping from alerts to entities, rather than updaters simply translating alerts to the internal model and feeding them from all updaters into one centralized store (not SIRI or GTFS specific, not specific to any one feed). So I'm hoping it will become unnecessary and won't be kept in the long term.

Reply:
I updated the Javadoc with your suggested names, "if kept".